### PR TITLE
Draft publish action

### PR DESCRIPTION
**Pull request by ppanero**
- partly closes https://github.com/zenodo/zenodo-rdm/issues/428

context from the tx fixture:

```
parent: 12132089
record: 12132090
draft: 12132091
# pid ops
pidstore_pid recid K UPDATE <- draft
pidstore_pid recid R UPDATE <- draft
pidstore_pid doi K INSERT
pidstore_pid oai R INSERT
pidstore_pid doi K INSERT

# bucket/file ops
files_files readable !writable UPDATE 
file_bucket locked UPDATE  -> draft bucket
file_bucket !locked INSERT -> record bucket
file_bucket !locked UPDATE  -> record bucket
files_object link_rec bucket INSERT
file_bucket +size UPDATE  -> record bucket
files_files readable !writable UPDATE 
files_object file_id bucket INSERT

# record/draft ops
records_metadata INSERT
records_buckets INSERT
communities_community_record INSERT  <- one per community
records_metadata UPDATE

# pid ops
pidstore_pid recid R UPDATE <- parent
pidstore_pid recid R object_uuid UPDATE <- draft
pidstore_redirect INSERT <- to record
pidstore_pid recid M UPDATE <- parent
pidrelations_pidrelation DELETE

# final record update
records_metadata UPDATE
```

### CODE REVIEW

File: migrator/zenodo_rdm_migrator/actions/transform/drafts.py [hunk 1]
```diff
@@ -145,3 +145,88 @@ def _transform_data(self):
...
+        # oai
+        #   draft 1 INSERT, final state R
+
+        # at the moment we only check operations over pids
+        # should be enough to differentiate between different types of publishing
+        # the operations over files/buckets can vary depending on the number of files
+        # the same happens with records and number of communities, etc.
+        return recid_rules
+
+    def _transform_data(self):  # pragma: no cover
```
Comment (ppanero) [inline migrator/zenodo_rdm_migrator/actions/transform/drafts.py:216]: ⚠️ did not get time to code this part but should be fairly straight forward `ZenodoDraftEntry` and I've also abstracted the parent to `ParentRecordEntry`. The PIDs is where it gets complex;

`DraftCreateAction` and `DraftEditAction` are inconsistent. The first one creates the PIDs and the inner denormalization (i.e. `draft.json.pid` and the respective one for the parent). While the second action class lets the `LoadAction` be the one that creates it.

On the one hand, I agree with what was discussed in another PR. It should be the `Load` who does it. However, that complicates the usage of split `pids` in the data payload. This means, for example, if we do not generate the pk and other values here in the transform then we would pass `{}` for most of the PIDs.

File: migrator/zenodo_rdm_migrator/actions/transform/drafts.py [hunk 2]
```diff
@@ -93,6 +93,7 @@ def _transform_data(self):
                 draft, parent = self._draft_and_parent_from_op(operation["after"])
 
         # calculate pids
+        # FIXME: undo this, move to load
```
Comment (ppanero) [inline migrator/zenodo_rdm_migrator/actions/transform/drafts.py:96]: this as discussed, didnt get time to undo

File: migrator/zenodo_rdm_migrator/actions/transform/drafts.py [hunk 3]
```diff
@@ -145,3 +146,142 @@ def _transform_data(self):
...
+        # should be enough to differentiate between different types of publishing
+        # the operations over files/buckets can vary depending on the number of files
+        # the same happens with records and number of communities, etc.
+        return recid_rules
+
+    def _transform_data(self):  # pragma: no cover
+        """Transforms the data and returns an instance of the mapped_cls."""
+        recids = set()
+        dois = set()
+        for operation in self.tx.operations:
```
Comment (ppanero) [inline migrator/zenodo_rdm_migrator/actions/transform/drafts.py:221]: _This code should work ™️_. However, it is not optimal, there is plenty of room to refactor/cleanup but the idea of how it is implemented is in a comment below.

**Comment by slint** [comment]
Included in #464.

**Pull request closed without merging**